### PR TITLE
test: fix android build test

### PR DESCRIPTION
**Pull request by zchenwei**
#### Description of changes

- Expo release a new major version a few days ago which changes the default server port. We need to specify a fixed port to the start command instead of relying on the default which is subject to change.
- Add an extra command to ensure dependencies are compatible with expo version.
- Bump the wait time for logging.
- Clean up some dead code and old comments.

Note: there may be some other flakiness that I did not catch but the above mentioned ones are the most obvious.

#### Description of how you validated changes

POC run succeeded five times in a row

- https://github.com/zchenwei/amplify-ui/actions/runs/5686585376
- https://github.com/zchenwei/amplify-ui/actions/runs/5686585107
- https://github.com/zchenwei/amplify-ui/actions/runs/5686584116
- https://github.com/zchenwei/amplify-ui/actions/runs/5686583767
- https://github.com/zchenwei/amplify-ui/actions/runs/5686583511

#### Checklist

- [x] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [x] PR description included
- [ ] `yarn test` passes and tests are updated/added
- [x] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: build-system-tests/scripts/mega-app-copy-files.sh [hunk 1]
```diff
@@ -188,14 +188,6 @@ fi
 if [[ "$FRAMEWORK" == "react-native" ]]; then
     echo "cp templates/components/react-native/cli/App.tsx mega-apps/${MEGA_APP_NAME}/App.tsx"
     cp templates/components/react-native/cli/App.tsx mega-apps/${MEGA_APP_NAME}/App.tsx
-    if [[ "$BUILD_TOOL" == 'cli' ]]; then
```
Comment (zchenwei) [inline build-system-tests/scripts/mega-app-copy-files.sh:191]: These 2 branches are running the same code and the comments are outdated.

File: build-system-tests/scripts/mega-app-copy-files.sh [hunk 2]
```diff
@@ -69,7 +69,7 @@ echo "#######################"
 echo "# Start Copying Files #"
 echo "#######################"
 
-if [[ "$FRAMEWORK" == "react-native" && "$BUILD_TOOL" == 'cli' ]]; then
+if [ "$FRAMEWORK" == "react-native" ]; then
```
Comment (zchenwei) [inline build-system-tests/scripts/mega-app-copy-files.sh:72]: This should be updated accordingly

File: build-system-tests/scripts/mega-app-install.sh [hunk 3]
```diff
@@ -137,6 +137,10 @@ else
     if [[ "$FRAMEWORK" == "react-native" ]]; then
         echo "npm install @aws-amplify/ui-react-native aws-amplify react-native-safe-area-context amazon-cognito-identity-js @react-native-community/netinfo @react-native-async-storage/async-storage react-native-get-random-values react-native-url-polyfill"
         npm install @aws-amplify/ui-react-native aws-amplify react-native-safe-area-context amazon-cognito-identity-js @react-native-community/netinfo @react-native-async-storage/async-storage react-native-get-random-values react-native-url-polyfill
+        if [[ "$BUILD_TOOL" == "expo" ]]; then
```
Comment (zchenwei) [inline build-system-tests/scripts/mega-app-install.sh:140]: This ensures dependencies are compatible with expo

File: build-system-tests/scripts/mega-app-install.sh [hunk 4]
```diff
@@ -137,6 +137,10 @@ else
...
+            echo "npx expo install --fix"
```
Comment (zchenwei) [inline build-system-tests/scripts/mega-app-install.sh:141]: This ensures dependencies are compatible with expo

File: build-system-tests/scripts/build-android.sh [hunk 5]
```diff
@@ -27,9 +27,9 @@ fi
 
 # Run npm run android in the background
 if [ $BUILD_TOOL == 'expo' ]; then
-  log "command" "npm run android"
+  log "command" "npm run android -- -p 19000"
   # Run npm run android in the background
-  npm run android &
+  npm run android -- -p 19000 &
```
Comment (zchenwei) [inline build-system-tests/scripts/build-android.sh:32]: We should specify a port instead of relying on the default

**Review by reesscot — APPROVED** [review]
LGTM, thanks for fixing this!

**Review by dbanksdesign — APPROVED** [review]

**Review by ioanabrooks — APPROVED** [review]

**Pull request merged**